src/frontend/src/handler/create_mv.rs

```diff
@@ -103,6 +109,10 @@ pub async fn handle_create_mv(
         (table, graph)
     };
 
+    if table.owner != *session.user_name() {
+        check_privilege(&session, &Object::TableId(table.id), Action::Select)?;
```

Comment:
This's wrong, you should check `Action::Select` in query relation rather than current creating table.